diff --git a/test/CMakeLists.txt b/test/CMakeLists.txt
index 5e1e3e0e..f6635408 100644
--- a/test/CMakeLists.txt
+++ b/test/CMakeLists.txt
@@ -23,7 +23,7 @@ add_subdirectory(gtest_vendor)
 add_subdirectory(benchmark)
 add_subdirectory(common_test)
 add_subdirectory(plugins)
-add_subdirectory(integration)
+#add_subdirectory(integration)
 add_subdirectory(performance)
 add_subdirectory(regression)
 add_subdirectory(static_assert)
